geoutils/georaster.py

```diff
@@ -1167,20 +1170,44 @@ def coords(self, offset='corner', grid=True):
         else:
             return xx[:-1], yy[:-1]
 
-    def xy2ij(self,x,y):
+    def xy2ij(self,x,y, op=math.floor,shift_area_or_point=False):
         """
         Return row, column indices for a given x,y coordinate pair.
 
         :param x: x coordinates
         :type x: array-like
         :param y: y coordinates
         :type y: array-like
+        :param op: operator to calculate index
+        :type op: operator
+        :param shift_area_or_point: shift according to AREA_OR_POINT GDAL attribute:
+         interpretation to what the raster value corresponds to (AREA = lower left or POINT = center)
 
         :returns i, j: indices of x,y in the image.
         :rtype i, j: array-like
 
         """
-        i, j = self.ds.index(x,y)
+        i, j = self.ds.index(x,y,op=op)
+
+        # # necessary because rio.Dataset.index does not return Iterable for a single point
+        if not isinstance(i, Iterable):
+            i, j = (np.asarray([i, ]), np.asarray([j, ]))
+        else:
+            i, j = (np.asarray(i), np.asarray(j))
+
+        # AREA_OR_POINT GDAL attribute, i.e. does the value refer to the lower left corner (AREA) or the center of pixel (POINT)
+        # This has no influence on georeferencing, it's only about the interpretation of the raster values, and thus only
+        # affects sub-pixel interpolation
+
+        if shift_area_or_point:
```

Comment:
It's a bit strange to have the area_or_point info but only use it in rare cases. Why not have an option 'area_or_point', which could be set to area, point or None. In the latter case, it would be set to the value in the metadata.
If you want to keep the same as current, you can just set default to 'area'.

Reply:
Done in 
903c260